sshkey.c

```diff
@@ -4661,6 +4661,43 @@ sshkey_parse_private_pem_fileblob(struct sshbuf *blob, int type,
 			sshkey_dump_ec_key(prv->ecdsa);
 # endif
 #endif /* OPENSSL_HAS_ECC */
+#ifdef HAVE_EVP_PKEY_GET_RAW_PUBLIC_KEY
+	} else if ((EVP_PKEY_base_id(pk) == EVP_PKEY_X25519
```

Comment:
I am not sure if we should accept the `EVP_PKEY_X25519` as Ed key -- they are similar, but can not be used for signing if I am right (unless there is something OpenSSL does with the keys to have this base_id).

Reply:
Thanks! Fixed, will be pushed soon